### PR TITLE
Ability to change granularity/rollups

### PR DESCRIPTION
**Pull request by Dalje-et**
### What
Implement a feature to allow the Grafana user to specify a value for granularity (for application/website) and rollUp (for infrastructure).

### Why
To make the Instana <-> Grafana plugin more customisable.

### CODE REVIEW

File: src/query_ctrl.ts [hunk 1]
```diff
@@ -384,12 +388,14 @@ export class InstanaQueryCtrl extends QueryCtrl {
...
     this.target.showGroupBySecondLevel = null;
     this.target.groupbyTagSecondLevelKey = null;
+    this.target.granularity = null;
     this.target.filters = [];
   }
 
   resetMetricSelection() {
     this.target.metric = null;
     this.target.filter = null;
+    this.target.granularity = null;
```
Comment (schmidtl) [inline src/query_ctrl.ts]: also reset rollup

File: src/datasource_website.ts [hunk 2]
```diff
@@ -129,19 +117,29 @@ export default class InstanaWebsiteDataSource extends AbstractDatasource {
...
-      metric['granularity'] = this.getChartGranularity(windowSize);
+      if (target.granularity) {
+        granularity = target.granularity;
+      } else {
+        granularity = getChartGranularity(windowSize, this.maximumNumberOfUsefulDataPoints);
+        target.granularity = granularity;
+      }
     }
 
+    metric['granularity'] = granularity.value;
```
Comment (schmidtl) [inline src/datasource_website.ts]: granularity might be null for if target.pluginId === "singlestat"

File: src/datasource_application.ts [hunk 3]
```diff
@@ -148,18 +131,27 @@ export default class InstanaApplicationDataSource extends AbstractDatasource {
...
-      metric['granularity'] = this.getChartGranularity(windowSize);
+      if (target.granularity) {
+        granularity = target.granularity;
+      } else {
+        granularity = getChartGranularity(windowSize, this.maximumNumberOfUsefulDataPoints);
+        target.granularity = granularity;
+      }
     }
 
+    metric['granularity'] = granularity.value;
```
Comment (schmidtl) [inline src/datasource_application.ts]: granularity might be null for if target.pluginId === "singlestat"

**Comment by schmidtl** [comment]
please adjust rollup retention for 1s to 24h as this is supported since 147

**Comment by Dalje-et** [comment]
Moved to #35

**Pull request closed without merging**